__tests__/jest-sandbox.test.js

```diff
@@ -66,3 +66,30 @@ test('Resets all mocks', () => {
   expect(x).toBeUndefined();
   expect(y).toBeUndefined();
 });
+
+test('Restores all spies', () => {
+  const sandbox = new JestSandbox();
+  class CoolClass {
+    hello() {
+      return 'hello';
+    }
+  }
+  const fn = sandbox.fn();
+  const spy = sandbox
+    .spyOn(CoolClass.prototype, 'hello')
+    .mockImplementation(() => 'yo, homie');
+  fn('a');
+  expect(fn).toBeCalledWith('a');
+  expect(fn.mock.calls).toHaveLength(1);
+  expect(fn.mock.instances).toHaveLength(1);
+  const cool = new CoolClass();
+  expect(cool.hello()).toBe('yo, homie');
+  expect(spy.mock.calls).toHaveLength(1);
+  expect(spy.mock.instances).toHaveLength(1);
+  sandbox.restore();
```

Comment:
@chmanie Thank you for your feedback! Yes, I can remove the lines 77, 82-84. Though, according to [the jest docs](https://facebook.github.io/jest/docs/en/mock-function-api.html#mockfnmockrestore):
> Beware that mockFn.mockRestore only works when mock was created with jest.spyOn. Thus you have to take care of restoration yourself when manually assigning jest.fn().

That's why I decided to test that it doesn't crash to call restore() when we have an `fn` in the sandbox.

The lines 87-88, 93-94 assert that `mockRestore` works as expected i.e. it [doesn't clear calls and instances](https://facebook.github.io/jest/docs/en/mock-function-api.html#mockfnmockclear):
> mockFn.mockClear()
Resets all information stored in the mockFn.mock.calls and mockFn.mock.instances arrays.

I can remove these lines either.

What do you think?